SourceCode/AgOpenGPS.Core/ApplicationModels/BaseModels/GeoCoord.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+
+namespace AgOpenGPS.Core
+{
+    public class GeoCoord
+    {
+        private double _northing;
+        private double _easting;
+
+        public GeoCoord(double northing, double easting)
+        {
+            _northing = northing;
+            _easting = easting;
+        }
+
+        public double Northing => _northing;
+        public double Easting => _easting;
+
+        public double DistanceSquared(GeoCoord coord2)
+        {
+            return new GeoDelta(this, coord2).LengthSquared;
+        }
+
+        public double Distance(GeoCoord coord2)
+        {
+            return Math.Sqrt(DistanceSquared(coord2));
```

Comment:
Maybe something like this would be more obvious:
```suggestion
            return (coord2 - this).Length;
```
?

Reply:
Changed to
            return new GeoDelta(this, coord2).Length;
to be more in line with method above.